File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/plugins/angularjs/service/specificationService.js
app.service('specificationService', function ($http) {

    this.findByPage = function (page, size, searchEntity) {
        return $http.post('../specification/findByPage.do?page=' + page + '&size=' + size, searchEntity);
    };

    this.findById = function (id) {
        return $http.get('../specification/findById.do?id=' + id);
    };

    this.save = function (entity) {
        return $http.post('../specification/save.do', entity);
    };

    this.del = function (ids) {
        return $http.get('../specification/del.do?ids=' + ids);
    };

    this.getOptionList = function () {
        return $http.get('../specification/getOptionList.do');
    };
});

File: pinyougou-parent/pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/SpecificationService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.pojo.TbSpecification;
import com.pinyougou.vo.SpecificationVo;
import entity.PageResult;

import java.util.List;
import java.util.Map;

public interface SpecificationService {

    PageResult<TbSpecification> findByPage(TbSpecification specification, int pageNum, int pageSize) throws Exception;

    SpecificationVo findBySpecificationId(Long id) throws Exception;

    void save(SpecificationVo specificationVo) throws Exception;

    void delBySpecificationIds(Long[] ids) throws Exception;

    List<Map<String, Object>> getOptionList() throws Exception;
}
